### PR TITLE
Move to jaas tag tests

### PR DESCRIPTION
**Pull request by kian99**
## Description

The `ToJAASTag` method was moved from the jujuapi to the jimm in https://github.com/canonical/jimm/pull/1122 but its test was not moved. This PR moves the test.
Additionally we add tests for roles.

Builds on #1466.

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [x] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

**Review by SimoneDutto — APPROVED** [review]
lgtm

File: internal/jimm/access.go [hunk 1]
```diff
@@ -532,6 +541,25 @@ func (t *tagResolver) groupTag(ctx context.Context, db *db.Database) (*ofga.Enti
...
+		zap.String("role-name", t.trailer),
+	)
+	if t.resourceUUID != "" {
+		return ofganames.ConvertTagWithRelation(jimmnames.NewRoleTag(t.resourceUUID), t.relation), nil
+	}
+	entry := dbmodel.RoleEntry{Name: t.trailer}
+
+	err := db.GetRole(ctx, &entry)
+	if err != nil {
+		return nil, errors.E(fmt.Sprintf("role %s not found", t.trailer))
```
Comment (alesstimec) [inline internal/jimm/access.go:557]: assumes a not found error..

File: internal/jimm/access_test.go [hunk 2]
```diff
@@ -495,7 +495,7 @@ func TestResolveTags(t *testing.T) {
 	err := j.Database.Migrate(ctx, false)
 	c.Assert(err, qt.IsNil)
 
-	identity, group, controller, model, offer, cloud, _ := createTestControllerEnvironment(ctx, c, j.Database)
+	identity, group, controller, model, offer, cloud, _, role := createTestControllerEnvironment(ctx, c, j.Database)
```
Comment (alesstimec) [inline internal/jimm/access_test.go:498]: yet another thing that is returned.. we really should return a struct to make this a bit easier on the eyes.. fine for a followup..

**Review by alesstimec — APPROVED** [review]
LGTM with one note

File: internal/jimm/access.go (hunk 1 above)
Comment (kian99) [reply internal/jimm/access.go:557]: True... Let me address this in a follow up.

**Pull request merged**